src/sonic-framework/tests/Makefile.am

```diff
@@ -0,0 +1,57 @@
+AUTOMAKE_OPTIONS = subdir-objects
+INCLUDES = -I $(top_srcdir) -I $(top_srcdir)/rebootbackend -I$(top_srcdir)/build/gen -I$(top_srcdir)/build/gen/github.com/openconfig/gnoi -I ../../sonic-swss-common/common
+
+TESTS = tests tests_asan tests_tsan tests_usan
+
+noinst_PROGRAMS = tests tests_asan tests_tsan tests_usan
+
+if DEBUG
+DBGFLAGS = -ggdb -DDEBUG
+else
+DBGFLAGS = -g -DNDEBUG
+endif
+
+CFLAGS_GTEST =
+LDADD_GTEST = -lgtest -lgtest_main -lgmock -lgmock_main
+CFLAGS_COVERAGE = --coverage -fprofile-arcs -ftest-coverage
+LDADD_COVERAGE = -lgcov
+CFLAGS_ASAN = -fsanitize=address
+CFLAGS_TSAN = -fsanitize=thread
+CFLAGS_USAN = -fsanitize=undefined
+
+testsdir = /usr/include
+
+tests_HEADERS = /usr/include/gmock/gmock.h
+
+tests_SOURCES = rebootbe_test.cpp \
+                $(top_srcdir)/rebootbackend/rebootbe.cpp \
+                reboot_thread_test.cpp \
+                $(top_srcdir)/rebootbackend/reboot_thread.cpp \
+                test_main.cpp
+
+tests_CFLAGS = $(DBGFLAGS) $(AM_CFLAGS) $(CFLAGS_COMMON) $(CFLAGS_GTEST) $(CFLAGS_COVERAGE) $(CFLAGS_SAI)
+tests_CPPFLAGS = $(DBGFLAGS) $(AM_CFLAGS) $(CFLAGS_COMMON) $(CFLAGS_GTEST) $(CFLAGS_COVERAGE) $(CFLAGS_SAI)
```

Comment:
>COVERAGE [](http://example.com/codeflow?start=84&length=8)

Could you share the coverage data? even better, you can refine build pipeline to expose the coverage data.

Reply:
Our internal tools show 90% coverage, we will look at refining the build pipeline to expose this as a followup item.